tests/unittests/tests-core/tests-core-cib.c

```diff
@@ -69,8 +68,53 @@ static void test_cib_peek__overflow(void)
     cib.read_count = UINT_MAX;
     cib.write_count = UINT_MAX;
 
-    TEST_ASSERT_EQUAL_INT(3, cib_put(&cib));
-    TEST_ASSERT_EQUAL_INT(3, cib_peek(&cib));
+    TEST_ASSERT_EQUAL_INT(TEST_CIB_SIZE-1, cib_put(&cib));
+    TEST_ASSERT_EQUAL_INT(TEST_CIB_SIZE-1, cib_peek(&cib));
+}
+
+static void test_cib_peek_at(void)
+{
+    TEST_ASSERT_EQUAL_INT(-1, cib_peek_at(&cib, 0));
```

Comment:
Could you add comments to the test cases that explain which behavior they cover / which behavior is expected?
Bonus points for adding the trivial test of:
```C
/* Test to get the next item */
TEST_ASSERT_EQUAL_INT( 0, cib_peek_at(&cib, 0));
/* This is exactly the same as using: */
TEST_ASSERT_EQUAL_INT( 0, cib_peek(&cib));
```
Just as tiny example + API coverage.

Reply:
Made the change via a fixup commit. I will squash when you are ready.

Reply:
Nice work!

Reply:
>  I will squash when you are ready

It's time!